// reason-apollo/src/ReasonApollo.bs.js
// Generated by BUCKLESCRIPT, PLEASE EDIT WITH CARE

import * as ApolloClient from "apollo-client";
import * as ReasonApolloQuery from "./graphql-types/ReasonApolloQuery.bs.js";
import * as ReasonApolloMutation from "./graphql-types/ReasonApolloMutation.bs.js";
import * as ReasonApolloSubscription from "./graphql-types/ReasonApolloSubscription.bs.js";

function createApolloClient(link, cache, ssrMode, ssrForceFetchDelay, connectToDevTools, queryDeduplication, param) {
  return new ApolloClient.ApolloClient({
              link: link,
              cache: cache,
              ssrMode: ssrMode,
              ssrForceFetchDelay: ssrForceFetchDelay,
              connectToDevTools: connectToDevTools,
              queryDeduplication: queryDeduplication
            });
}

var CreateQuery = ReasonApolloQuery.Make;

var CreateMutation = ReasonApolloMutation.Make;

var CreateSubscription = ReasonApolloSubscription.Make;

var Provider;

var Consumer;

export {
  createApolloClient ,
  CreateQuery ,
  CreateMutation ,
  CreateSubscription ,
  Provider ,
  Consumer ,
  
}
/* apollo-client Not a pure module */
