// bs-axios/src/axios.js
// Generated by BUCKLESCRIPT, PLEASE EDIT WITH CARE


function makeResponseTransformer1(f) {
  return [f];
}

function makeResponseTransformer2(f0, f1) {
  return /* tuple */[
          f0,
          f1
        ];
}

function makeResponseTransformer3(f0, f1, f2) {
  return /* tuple */[
          f0,
          f1,
          f2
        ];
}

function makeResponseTransformer4(f0, f1, f2, f3) {
  return /* tuple */[
          f0,
          f1,
          f2,
          f3
        ];
}

function makeResponseTransformer5(f0, f1, f2, f3, f4) {
  return /* tuple */[
          f0,
          f1,
          f2,
          f3,
          f4
        ];
}

function makeRequestTransformer1(f) {
  return [f];
}

function makeRequestTransformer2(f0, f1) {
  return /* tuple */[
          f0,
          f1
        ];
}

function makeRequestTransformer3(f0, f1, f2) {
  return /* tuple */[
          f0,
          f1,
          f2
        ];
}

function makeRequestTransformer4(f0, f1, f2, f3) {
  return /* tuple */[
          f0,
          f1,
          f2,
          f3
        ];
}

function makeRequestTransformer5(f0, f1, f2, f3, f5) {
  return /* tuple */[
          f0,
          f1,
          f2,
          f3,
          f5
        ];
}

function Headers_fromObj(prim) {
  return prim;
}

function Headers_fromDict(prim) {
  return prim;
}

var $$Headers = {
  fromObj: Headers_fromObj,
  fromDict: Headers_fromDict
};

var Instance;

var Agent;

export {
  $$Headers ,
  makeResponseTransformer1 ,
  makeResponseTransformer2 ,
  makeResponseTransformer3 ,
  makeResponseTransformer4 ,
  makeResponseTransformer5 ,
  makeRequestTransformer1 ,
  makeRequestTransformer2 ,
  makeRequestTransformer3 ,
  makeRequestTransformer4 ,
  makeRequestTransformer5 ,
  Instance ,
  Agent ,
  
}
/* No side effect */
